### PR TITLE
Support tooltip rendering in the prediction list view

### LINKED ISSUE #3638: Support rendering the tooltip in PSReadLine list view
### Prerequisites

- [X] Write a descriptive title.

### Description of the new feature/enhancement

Today, the `PredictiveSuggestion` type has the `Tooltip` property, but PSReadLine doesn't use it in its rendering.
For completeness, it should render the tooltip, and ideally the tooltips should be displayed in a popup terminal UI window when the user moves the focus to an item in the list, in the form similar to this: https://github.com/waf/CSharpRepl#c-repl

### Proposed technical implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by daxian-dbw**
# PR Summary

Support tooltip rendering in the prediction list view. Fix #3638.

- By default, the list view can use up to 4 lines for tooltip rendering:

   tooltip1

- For tooltip that spans more than 4 lines, we allow the user to see the full tooltip in the alternate screen buffer with the keybinding <kbd>F4</kbd>

   tooltip2

- When terminal size shrinks, the maximum lines can be used for tooltip also decreases -- it uses up to 2 lines for tooltip when the max list height is reduced to 5.

   tooltip3

- When the terminal size is even smaller and the max list view height becomes 3, it uses up to 1 line for tooltip.

   tooltip4

- You can change the tooltip color by `Set-PsreadLineOption -Colors @{ ListPredictionTooltip = "green" }`, for example.

   tooltip5

## PR Checklist

- [x] PR has a meaningful title
    - Use the present tense and imperative mood when describing your changes
- [x] Summarized changes
- [x] Make sure you've added one or more new tests
- [x] Make sure you've tested these changes in terminals that PowerShell is commonly used in (i.e. conhost.exe, Windows Terminal, Visual Studio Code Integrated Terminal, etc.)
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] Documentation needed at [PowerShell-Docs](https://github.com/MicrosoftDocs/PowerShell-Docs)
        - [x] Doc Issue filed: 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/PowerShell/PSReadLine/pull/3667)

### CODE REVIEW

**Comment by StevenBucher98** [comment]
Love the UI! Is there a way to disable the tooltip?

**Comment by daxian-dbw** [comment]
Same as disabling tooltip in menu completion today: `Set-PSReadLineOption -showToolTips:$false`

**Review by StevenBucher98 — APPROVED** [review]
Can't give a proper code review but the design and UX look great to me!

File: PSReadLine/Prediction.Views.cs [hunk 1]
```diff
@@ -865,6 +917,146 @@ static StringBuilder AppendColor(StringBuilder buffer, string colorToUse, ref st
...
             }
 
+            /// <summary>
+            /// Generate the rendering text for the tooltip.
+            /// </summary>
+            private int RenderTooltip(string tooltip, List<StringBuilder> consoleBufferLines, ref int currentLogicalLine)
+            {
+                const int LengthOfLeadingPart = 6;
+                const string IndicatorSymbol = ">>";
+                const string MsgForViewAll = "(<F4> to view all)";
```
Comment (andyleejordan) [inline PSReadLine/Prediction.Views.cs:927]: Just curious if we do this elsewhere. I suspect that key binding is customizable by the user, but this won't update. Is that the case?

File: test/ListViewTooltipTest.cs
Comment (andyleejordan) [inline test/ListViewTooltipTest.cs:1]: Nice test.

**Review by andyleejordan — APPROVED** [review]
Looks good to me!

File: PSReadLine/Prediction.Views.cs (hunk 1 above)
Comment (daxian-dbw) [reply PSReadLine/Prediction.Views.cs:927]: Right, the <F4> key could already be bound by a user and this message currently won't update in that case.
We don't do this elsewhere -- first time advertising a key-binding in the rendering. Will collect the feedback and see how to best deal with it.

Comment (daxian-dbw) [reply PSReadLine/Prediction.Views.cs:927]: Opened #3668 to track this.

**Pull request merged**